common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -530,6 +540,65 @@ protected SecretKey loadSecretKey(@NonNull final KeyType keyType) throws IOExcep
         throw new GeneralSecurityException(ErrorStrings.UNKNOWN_ERROR);
     }
 
+    /**
+     * If needed, get the key from inactive broker (if there is one).
+     * If it fails to get a new key. It will create a new one.
+     *
+     * This function can only be invoked by Broker apps.
+     */
+    public void migrateEncryptionKeyIfNeeded() throws GeneralSecurityException, IOException {
+        final String methodName = ":migrateEncryptionKeyIfNeeded";
+
+        final String activeBrokerPackageName = getPackageName();
+
+        if (!(COMPANY_PORTAL_APP_PACKAGE_NAME.equalsIgnoreCase(activeBrokerPackageName) ||
+                AZURE_AUTHENTICATOR_APP_PACKAGE_NAME.equalsIgnoreCase(activeBrokerPackageName))) {
+            final String errorMessage = "Caller is not a broker. Migration is not needed";
+            Logger.error(TAG + methodName, errorMessage, null);
+            throw new InvalidParameterException(errorMessage);
```

Comment:
This method does not accept any parameters yet we throw an `InvalidParameterException`. May be we can throw some other exception that will be better suited here. I'll see if I can find any existing java exception that will fit our needs.

Reply:
Good alternatives here might be `IllegalStateException` or `UnsupportedOperationException`

Reply:
yes `IllegalStateException` might be good here